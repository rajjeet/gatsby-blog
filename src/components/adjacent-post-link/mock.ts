export const previous = {
  fields: { slug: '/previous/' },
  frontmatter: { title: 'Introducing the New React Concurrent Mode' },
};
export const next = {
  fields: { slug: '/next/' },
  frontmatter: { title: 'The Implications of React Concurrent Mode' },
};
