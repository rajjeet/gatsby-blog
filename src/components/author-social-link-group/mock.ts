import { TProps } from './types';

export const makeProps = (): TProps => ({
  className: '',
});

