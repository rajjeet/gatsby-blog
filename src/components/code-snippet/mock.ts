import { TProps } from './types';

export const makeProps = (): TProps => ({
  children: `
const mainReducer = (state, action) => {
  console.log('Before', state);
  console.log('Action', action);
  let result = {
    count: countReducer(state.count, action)      
  };
  console.log('After', result);
  return result
};      
`,
  fileName: 'index.js',
});
