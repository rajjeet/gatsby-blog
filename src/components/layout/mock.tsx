import React from 'react';
import { TProps } from './types';
import { createMockGatsbyImageSharpFluid } from '../../utils/testing';
import { siteMetadata } from '../../../gatsby-config';

export const makeProps = (): TProps => ({
  children: <div>Sample child</div>,
  data: {
    file: createMockGatsbyImageSharpFluid.file,
    site: siteMetadata,
  },
});

