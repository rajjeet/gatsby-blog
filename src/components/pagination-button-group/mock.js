export const makeProps = () => ({
  className: {}, currentPage: 1, numOfPages: 5, paginationSlug: '',
});
