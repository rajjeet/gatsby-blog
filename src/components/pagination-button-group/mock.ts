export const makeProps = ({ currentPage = 1, numOfPages = 5 } = {}): TProps => ({
  className: '', currentPage, numOfPages, paginationSlug: '',
});

export type TProps = {
  className: string;
  currentPage: number;
  numOfPages: number;
  paginationSlug: string;
};
