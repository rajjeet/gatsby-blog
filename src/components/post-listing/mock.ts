import { TProps } from './types';
import { mockPosts } from '../../fixtures/posts';

export const makeProps = ({
  posts = mockPosts,
  heading = 'Post Listing',
  numOfPages = 5,
  currentPage = 1,
  paginationSlug = '/sample_slug/',
} = {}): TProps => ({
  posts,
  heading,
  numOfPages,
  currentPage,
  paginationSlug,
});

