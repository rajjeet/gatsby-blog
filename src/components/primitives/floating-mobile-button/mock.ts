import { faTimes } from '@fortawesome/free-solid-svg-icons';

export const makeProps = () => ({
  icon: faTimes,
});
