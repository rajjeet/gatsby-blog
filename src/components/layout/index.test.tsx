import React from 'react';
import { render } from '@testing-library/react';
import { Layout } from './index';
import { makeProps } from './mock';

describe('<Layout />', () => {
  it('should render', () => {
    const { asFragment } = render(<Layout {...makeProps()} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render the pass in children', () => {
    const { getByText } = render(<Layout {...makeProps()} />);
    expect(getByText('Sample child')).toBeDefined();
  });

  it('should render an image for website logo', () => {
    const { getByAltText } = render(<Layout {...makeProps()} />);
    expect(getByAltText(/ortmesh.*logo/i)).toBeDefined();
  });

  it('should have a link with label home', () => {
    const { getByLabelText } = render(<Layout {...makeProps()} />);
    expect(getByLabelText(/home/i)).toBeDefined();
  });

  it('should have heading called Ortmesh', () => {
    const { getByText } = render(<Layout {...makeProps()} />);
    expect(getByText('Ortmesh')).toBeDefined();
  });

  it('should have a website caption', () => {
    const { getByText } = render(<Layout {...makeProps()} />);
    expect(getByText('Write code that matters')).toBeDefined();
  });
});

