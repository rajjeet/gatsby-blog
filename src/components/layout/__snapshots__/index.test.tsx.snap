// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render 1`] = `
<DocumentFragment>
  <div
    class="sc-htpNat faJHDz"
  >
    <div
      class="sc-bwzfXH bmdiOC"
    >
      <a
        aria-label="See Home"
        href="/"
      >
        <div
          class=" gatsby-image-wrapper"
          style="position: relative; overflow: hidden; width: 3em;"
        >
          <div
            style="width: 100%; padding-bottom: 100%;"
          />
          <picture>
            <source
              sizes=""
              srcset=""
            />
            <img
              alt="Ortmesh logo"
              loading="lazy"
              sizes=""
              src="https://via.placeholder.com/250x180"
              srcset=""
              style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
            />
          </picture>
          <noscript>
            &lt;picture&gt;&lt;source srcset="" /&gt;&lt;img loading="lazy" src="https://via.placeholder.com/250x180" alt="Ortmesh logo" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
          </noscript>
        </div>
      </a>
      <a
        href="/"
      >
        <h2>
          Ortmesh
        </h2>
        <p>
          Write code that matters
        </p>
      </a>
      <button
        class="sc-bdVaJa eSxrIN"
      >
        About
      </button>
    </div>
    <div>
      Sample child
    </div>
  </div>
</DocumentFragment>
`;
